src/lightning/fabric/fabric.py

```diff
@@ -614,7 +617,17 @@ def load(
             The remaining items that were not restored into the given state dictionary. If no state dictionary is
             given, the full checkpoint will be returned.
         """
-        return self._strategy.load_checkpoint(path=path, state=state)
+        unwrapped_state = _unwrap_objects(state)
+        remainder = self._strategy.load_checkpoint(path=path, state=unwrapped_state)
+        # We need to unwrap objects (see above) but this creates a new dictionary. In-place updates
+        # (for user metadata) wouldn't show up in the original dict, so we need to copy the data back.
+        for k in list(unwrapped_state.keys()):
+            if isinstance(state[k], (_FabricModule, _FabricOptimizer, _FabricDataLoader)):  # type: ignore[index]
+                continue
+            state[k] = unwrapped_state[k]  # type: ignore[index]
```

Comment:
Mypy complained here because state can be optional, so you cannot index a None.

I think that's a valid concern